gazebo_ros/src/gazebo_ros_factory.cpp

```diff
@@ -349,10 +349,9 @@ void GazeboRosFactoryPrivate::AddNamespace(
 
         // Then we can add the element
         ns_elem = ros_elem->AddElement("namespace");
+        // Set namespace
+        ns_elem->Set<std::string>(_robot_namespace);
       }
-
-      // Set namespace
-      ns_elem->Set<std::string>(_robot_namespace);
```

Comment:
If the user sets a `robot_namespace` on the factory request, we do want to override the values within the SDF file. So I think this was correct originally. And we should already be skipping it if the user left it empty:

https://github.com/ros-simulation/gazebo_ros_pkgs/blob/57732fe150efcb324e905f0f68537888c1eceb30/gazebo_ros/src/gazebo_ros_factory.cpp#L231-L235

Reply:
The following condition will never be false. Because the deafult namespace is '/'

https://github.com/ros-simulation/gazebo_ros_pkgs/blob/57732fe150efcb324e905f0f68537888c1eceb30/gazebo_ros/src/gazebo_ros_factory.cpp#L233

Here is where the default value is set 

https://github.com/ros-simulation/gazebo_ros_pkgs/blob/57732fe150efcb324e905f0f68537888c1eceb30/gazebo_ros/scripts/spawn_entity.py#L73-L74

Reply:
Fixed and updated the title of the PR